tests/lib/bin/retry

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+MAX_ATTEMPTS=20
+SLEEP_TIME=.5
+for i in "$@"; do
+    case $i in
+        --max-attempts)
+        MAX_ATTEMPTS="$2"
+        shift
+        shift
+        ;;
+        --sleep-time)
+        SLEEP_TIME="$2"
+        shift
+        shift
+        ;;
+    esac
+done
+
+COMMAND="$@"
```

Comment:
This is slightly confusing but that's not what I meant:

You want to do this instead:
```
cmd="$1"; shift
"$cmd" "$@"
```

"$@" expands to multiple elements do using it with `test` like you did is incorrect.